core-java-modules/core-java-collections-list-6/src/test/java/com/baeldung/modifyandprint/ModifyAndPrintListElementsUnitTest.java

```diff
@@ -0,0 +1,56 @@
+package com.baeldung.modifyandprint;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.Lists;
+
+public class ModifyAndPrintListElementsUnitTest {
+
+    private final Logger log = LoggerFactory.getLogger(ModifyAndPrintListElementsUnitTest.class);
+
+    @Test
+    void whenPrintingInForEach_thenListIsPrinted() {
+        List<String> theList = Lists.newArrayList("Kai", "Liam", "Eric", "Kevin");
+        theList.forEach(e -> log.info(e));
+    }
+
+    @Test
+    void whenUsingModifyAndPrintingSeparately_thenListIsModifiedAndPrinted() {
+        List<String> theList = Lists.newArrayList("Kai", "Liam", "Eric", "Kevin");
+        theList.replaceAll(e -> e.toUpperCase());
+        theList.forEach(e -> log.info(e));
+        assertEquals(List.of("KAI", "LIAM", "ERIC", "KEVIN"), theList);
+    }
+
+    @Test
+    void whenPrintingInMap_thenStreamIsModifiedAndPrinted() {
+        List<String> theList = List.of("Kai", "Liam", "Eric", "Kevin");
+        List<String> newList = theList.stream()
+            .map(e -> {
+                String newElement = e.toUpperCase();
+                log.info(newElement);
+                return newElement;
+            })
+            .filter(e -> e.length() == 4)
```

Comment:
To keep the focus of the article, I feel like we should drop the filters and just stick to the case where we are modifying all the elements

Reply:
filter() removed